src/System.Net.Http/tests/FunctionalTests/ManagedHandlerTest.cs

```diff
@@ -261,7 +263,52 @@ public override int Read(byte[] buffer, int offset, int count)
             public override long Seek(long offset, SeekOrigin origin) => throw new NotImplementedException();
             public override void SetLength(long value) => throw new NotImplementedException();
         }
+    }
+
+    public sealed class ManagedHandler_ConnectionUpgrade_Test : HttpClientTestBase
+    {
+        protected override bool UseManagedHandler => true;
+
+        [Fact]
+        public async Task UpgradeConnection_Success()
+        {
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            {
+                using (HttpClient client = CreateHttpClient())
+                {
+                    // We need to use ResponseHeadersRead here, otherwise we will hang trying to buffer the response body.
+                    Task<HttpResponseMessage> getResponseTask = client.GetAsync(url, HttpCompletionOption.ResponseHeadersRead);
+                    await LoopbackServer.AcceptSocketAsync(server, async (s, stream, reader, writer) =>
```

Comment:
Nit: it'd be nice to call these serverReader, serverWriter, etc., so that below it's clear when things are dealing with client vs server

Reply:
Will do